torch/fx/experimental/symbolic_shapes.py

```diff
@@ -341,7 +341,11 @@ def eval(cls, *args):
 @lru_cache(256)
 def safe_expand(r):
     if hasattr(r, 'expand'):
-        return sympy.expand(r)
+        try:
+            return sympy.expand(r)
+        except RecursionError:
+            log.warning(f"RecursionError in sympy.expand({r})")
+            return r
```

Comment:
Fix for `test/inductor/test_torchinductor.py::CpuTests::test_output_strides_dynamic_shapes_cpu`. We're doing this for sympy.solve as well, so seems fine, right?